internal/handler/messaging.go

```diff
@@ -173,10 +175,22 @@ func (h *Messaging) processMessageQueue(message mq.Message) {
 		if insights.InsightsType != Direct {
 			err := h.sendToLagoonS3(incoming, insights, resource)
 			if err != nil {
-				//log.Printf("Unable to send to S3: %s", err.Error())
-				slog.Error("Unable to send to S3", "Error", err.Error())
-
-				// TODO: BETTER ERROR HANDLING
+				incoming.RequeueAttempts++
+				updatedMessage, err := json.Marshal(incoming)
+				if err != nil {
+					fmt.Printf(err.Error())
+				}
+				if incoming.RequeueAttempts <= 3 {
```

Comment:
I wonder if this "3" should at the very least be a constant that's defined somewhere - maybe we don't need it to be an actual parameter, but a constant rather than a magic number would be good, I reckon?